tf2/include/tf2/time_cache.h

```diff
@@ -44,6 +44,14 @@ namespace tf2
 {
 typedef std::pair<tf2::TimePoint, tf2::CompactFrameID> P_TimeAndFrameID;
 
+enum class TimeCacheError : std::uint8_t
+{
+  TIME_CACHE_NO_ERROR = 0,
+  TIME_CACHE_NOT_ENOUGH_DATA_ERROR = 1,
+  TIME_CACHE_BACKWARD_EXTRAPOLATION_ERROR = 2,
+  TIME_CACHE_FORWARD_EXTRAPOLATION_ERROR = 3,
+};
```

Comment:
Is there any particular reason to introduce this enum here?  Since it looks like we are just going to convert it to a `TF2Error` at some point anyway, why not change all of the APIs to just have an out parameter of `TF2Error`?

Reply:
Well, in fact it would also simplify a bit the code... We basically just need to include here the exceptions.h header.

Reply:
Tomorrow I may already provide the patch 👍

Reply:
Thank you!